schema/parser.go

```diff
@@ -0,0 +1,94 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema // import "go.opentelemetry.io/otel/schema"
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"strings"
+
+	"golang.org/x/mod/semver"
+	"gopkg.in/yaml.v2"
+
+	"go.opentelemetry.io/otel/schema/ast"
+)
+
+func Parse(schemaFile string) (*ast.Schema, error) {
```

Comment:
Can this accept an `io.Reader` instead? It would mean this could be passed an HTTP response body or some other user implemented receiver.

Reply:
Yes, that's a good point.

Reply:
Done.